pkg/controllers/management/clusterupstreamrefresher/cluster_upstream_refresher.go

```diff
@@ -120,6 +126,8 @@ func getProviderRefreshInterval(provider string) (time.Duration, error) {
 	switch provider {
 	case apimgmtv3.ClusterDriverEKS:
 		refreshInterval = settings.EKSUpstreamRefresh.GetInt()
+	case apimgmtv3.ClusterDriverGKE:
```

Comment:
I think we should go with the suggestion here instead of having a switch in this function: https://github.com/rancher/rancher/pull/31939/files#diff-ec482fc6c00713153755d85cb92dd1a3f9c76b416d548446a5f5d5728d4afec0R129-R130

Reply:
@rmweir the link points back here

Reply:
But what if GKE and EKS settings will have different refresh times ?

Reply:
He is saying to do `getSettingById(fmt.sprintf("%s-refresh"), provider))` so it gets auto detected and pulled for each provider.

Reply:
according to our discussion, I have included suggestions